### PR TITLE
Added release workflow

### PR DESCRIPTION
**Pull request by jmoreira-valory**
Added release workflow

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        with:
+          python-version: ${{ matrix.python-versions }}
+      - uses: addnab/docker-run-action@v3
+        with:
+            image: valory/open-autonomy-user:latest
+            options: -v ${{ github.workspace }}:/work
+            run: |
+              echo "Pushing Packages"
+              cd /work
+              export AUTHOR=$(grep 'service' packages/packages.json | awk -F/ '{print $2}' | head -1)
```
Comment (DavidMinarsch) [inline .github/workflows/release.yaml:27]: Please confirm this yields the correct author here

File: .github/workflows/release.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+              if [ $? -eq 0 ]; then
+                  export TAG=`echo $TAG | sed 's/^v//'`
+              else
+                  echo "You are not on a tagged branch"
+                  exit 1
+              fi
+              echo VERSION=$TAG> env.sh
+              echo AUTHOR=$(grep 'service' packages/packages.json | awk -F/ '{print $2}' | head -1) >> env.sh
+              echo SERVICE=$(grep 'service' packages/packages.json | awk -F/ '{print $3}' | head -1) >> env.sh
+              echo DEFAULT_IMAGE_TAG=$(cat packages/packages.json | grep agent | awk -F: '{print $2}' | tr -d '", ') >> env.sh
```
Comment (DavidMinarsch) [inline .github/workflows/release.yaml:63]: same here

File: .github/workflows/release.yaml (hunk 1 above)
Comment (0xArdi) [reply .github/workflows/release.yaml:27]: this will be `balancer`, thats who the author of the package is.

File: .github/workflows/release.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+      - name: Docker login
+        run: |
+            echo  $DOCKER_PASSWORD | docker login -u $DOCKER_USER --password-stdin 
+      - name: Docker Push
+        run: |
+          source env.sh
+          echo "Pushing $DOCKER_USER/$SERVICE:$VERSION"
+          echo "Pushing $DOCKER_USER/$SERVICE:$DEFAULT_IMAGE_TAG"
+          docker push $DOCKER_USER/oar-$SERVICE:$VERSION
+          docker push $DOCKER_USER/oar-$SERVICE:$DEFAULT_IMAGE_TAG
```
Comment (0xArdi) [inline .github/workflows/release.yaml:92]: we have two services in the repo, only one of them will get a tag. Should both of them get tags?

File: .github/workflows/release.yaml (hunk 1 above)
Comment (jmoreira-valory) [reply .github/workflows/release.yaml:27]: Output:
AUTHOR=balancer

File: .github/workflows/release.yaml (hunk 2 above)
Comment (jmoreira-valory) [reply .github/workflows/release.yaml:63]: Output:
VERSION=TestTag
AUTHOR=balancer
SERVICE=autonomous_fund_goerli
DEFAULT_IMAGE_TAG=bafybeiaq6453bqemn2idlbo6bzs24exrbp7v2rzui45s3clgmihy6smdua

File: .github/workflows/release.yaml (hunk 3 above)
Comment (8ball030) [reply .github/workflows/release.yaml:92]: Correct me if im wrong, but the oar image is actuall the same for both services

Comment (8ball030) [reply .github/workflows/release.yaml:92]: so this hsould be the same name

Comment (8ball030) [reply .github/workflows/release.yaml:92]: image

@jose thanks for confirming

Comment (8ball030) [reply .github/workflows/release.yaml:92]: Otheerwise, we will need to put in a function, and then do the function for each service. 

I *think* we can do loops in actions,

**Review by 0xArdi — APPROVED** [review]

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**